relentless/optimize/objective.py

```diff
@@ -165,3 +168,66 @@ def design_variables(self):
         :class:`ObjectiveFunction` for which the result was constructed, mapped
         to the value of the variables at the time the result was constructed."""
         return self._design_variables
+
+class RelativeEntropy(ObjectiveFunction):
+    def __init__(self, target, simulation, potentials, thermo, dr, communicator=None):
+        if target.V is None or target.N is None:
+            raise ValueError('The target ensemble must have both V and N set.')
+
+        self.target = target
+        self.simulation = simulation
+        self.potentials = potentials
+        self.thermo = thermo
+        self.dr = dr
+        self.communicator = communicator
+
+    def compute(self, directory=None):
+        sim = self.simulation.run(self.target, self.potentials, directory, self.communicator)
+        sim_ens = self.thermo.extract_ensemble(sim)
+
+        # compute the relative entropy gradient by integration
+        g_tgt = self.target.rdf
+        g_sim = sim_ens.rdf
+        dvars = self.design_variables()
+        gradient = _collections.KeyedArray(keys=dvars)
+
+        for var in dvars:
+            update = 0
+            for i,j in self.target.pairs:
+                # evaluate derivative wrt design variable and interpolate with r
+                rs = self.potentials.pair.r
+                dus = self.potentials.pair.derivative((i,j),var)
```

Comment:
I would evaluate u directly, since this is what is going to force g(r) = 0, to check for infinity rather than the derivative.